### PR TITLE
Cyclops Horn Sync

### PR DESCRIPTION
**Pull request by Jannify**
Sync Cyclops Horn

### CODE REVIEW

**Comment by Jannify** [comment]
@MarijnS95 Ups... I will fix it.

File: NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+using System;
+using UnityEngine;
+
+namespace NitroxClient.Communication.Packets.Processors
+{
+    public class CyclopsActivateHornProcessor : ClientPacketProcessor<CyclopsActivateHorn>
+    {
+        private PacketSender packetSender;
+
+        public CyclopsActivateHornProcessor(PacketSender packetSender) {
```
Comment (Sunrunner37) [inline NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs]: Please match the style formatting done thoughout the application("Curly brace on a new line")

File: NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        public override void Process(CyclopsActivateHorn packet) {
+            Optional<GameObject> opCyclops = GuidHelper.GetObjectFrom(packet.Guid);
+
+            if (opCyclops.IsPresent()) {
+                CyclopsHornControl horn = opCyclops.Get().GetComponentInChildren<CyclopsHornControl>();
+
+                if (horn != null) {
+                    using (packetSender.Suppress<CyclopsActivateHorn>()) {
+                        Utils.PlayEnvSound(horn.hornSound, horn.hornSound.gameObject.transform.position, 20f);
+                    }
```
Comment (MarijnS95) [inline NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs]: You are patching `OnPress`, which calls `PlayEnvSound`, so there's no need to suppress the packet unless you'd call `OnPress` directly here.

**Comment by Mrtupu** [comment]
Jessesalmi  in vite

File: NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs (hunk 2 above)
Comment (Jannify) [reply NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs]: Done in new push (https://github.com/Sunrunner37/Nitrox/pull/23/files)

File: NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs (hunk 1 above)
Comment (Jannify) [reply NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs]: Done in new push (https://github.com/Sunrunner37/Nitrox/pull/23/files)

File: NitroxPatcher/NitroxPatcher.csproj [hunk 3]
```diff
@@ -51,10 +51,13 @@
     <Reference Include="System.Data.DataSetExtensions" />
     <Reference Include="System.Data" />
     <Reference Include="System.Xml" />
+    <Reference Include="UnityEngine.UI, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null" />
```
Comment (Sunrunner37) [inline NitroxPatcher/NitroxPatcher.csproj]: Please remove this reference.

File: NitroxClient/Communication/Packets/Processors/CyclopsBeginSilentRunningProcessor.cs [hunk 4]
```diff
@@ -31,13 +31,11 @@ public override void Process(CyclopsBeginSilentRunning packet)
                         ability.subRoot.BroadcastMessage("RigForSilentRunning");
                         ability.InvokeRepeating("SilentRunningIteration", 0f, ability.silentRunningIteration);
                     }
-                }
-                else
+                } else
```
Comment (Sunrunner37) [inline NitroxClient/Communication/Packets/Processors/CyclopsBeginSilentRunningProcessor.cs]: Please revert the changes in this file.

File: NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    public class CyclopsActivateHornProcessor : ClientPacketProcessor<CyclopsActivateHorn>
+    {
+        private PacketSender packetSender;
+
+        public CyclopsActivateHornProcessor(PacketSender packetSender)
+        {
+            this.packetSender = packetSender;
+        }
+
+        public override void Process(CyclopsActivateHorn hornPacket)
```
Comment (MarijnS95) [inline NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs]: IIRC we use the variable name `packet` in most if not all other packet processors, for consistency.

File: NitroxClient/Communication/Packets/Processors/CyclopsActivateShieldProcessor.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    public class CyclopsActivateShieldProcessor : ClientPacketProcessor<CyclopsActivateShield>
+    {
+        private PacketSender packetSender;
+
+        public CyclopsActivateShieldProcessor(PacketSender packetSender)
+        {
+            this.packetSender = packetSender;
+        }
+
+        public override void Process(CyclopsActivateShield shieldPacket)
```
Comment (MarijnS95) [inline NitroxClient/Communication/Packets/Processors/CyclopsActivateShieldProcessor.cs]: Idem.

File: NitroxClient/Communication/Packets/Processors/CyclopsBeginSilentRunningProcessor.cs [hunk 7]
```diff
@@ -40,7 +40,7 @@ public override void Process(CyclopsBeginSilentRunning packet)
             else
             {
                 Console.WriteLine("Could not find cyclops with guid " + packet.Guid + " to begin silent running.");
-            }
+}
```
Comment (MarijnS95) [inline NitroxClient/Communication/Packets/Processors/CyclopsBeginSilentRunningProcessor.cs]: Here should be indentation.

File: NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs (hunk 5 above)
Comment (Jannify) [reply NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs]: In CyclopsToggleFloodlightsProcessor the name is lightingPacket...

Comment (MarijnS95) [reply NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs]: You are right (And my IIRC is wrong :wink:). About 50% is named `packet` (because that is what VS names it when auto-implementing the abstract class), and the other 50% has a name related to the processor.

Comment (Jannify) [reply NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs]: Ok and what should I do now (and what means IIRC) ?

Comment (MarijnS95) [reply NitroxClient/Communication/Packets/Processors/CyclopsActivateHornProcessor.cs]: `If I Remember Correctly` - I wasn't completely sure in the first place.

You can just ignore this, it's very very minor anyway.
